shadows/framework/src/main/java/org/robolectric/shadows/ShadowBitmapFactory.java

```diff
@@ -147,24 +159,24 @@ protected static Bitmap decodeStream(
       // ignore
     }
 
-    String name = (is instanceof NamedStream)
-        ? is.toString().replace("stream for ", "")
-        : null;
-    Point imageSize = (is instanceof NamedStream) ? null : getImageSizeFromStream(is);
+    boolean isNamedStream = is instanceof NamedStream;
+    String name = isNamedStream ? is.toString().replace("stream for ", "") : null;
+    ImageUtil.RobolectricBufferedImage image = isNamedStream ? null : getImageFromStream(is);
+    BufferedImage bufferedImage = image == null ? null : image.bufferedImage;
+    Point imageSize =
+        bufferedImage == null
+            ? null
+            : new Point(bufferedImage.getWidth(), bufferedImage.getHeight());
     Bitmap bitmap = create(name, outPadding, opts, imageSize);
     ReflectionHelpers.callInstanceMethod(bitmap, "setNinePatchChunk",
             ClassParameter.from(byte[].class, ninePatchChunk));
     ShadowBitmap shadowBitmap = Shadow.extract(bitmap);
     shadowBitmap.createdFromStream = is;
 
-    try {
-      if (is != null && opts != null) {
-        is.reset();
-        opts.outMimeType = URLConnection.guessContentTypeFromStream(is);
-      }
-    } catch (IOException e) {
-      // ignore
+    if (image != null && opts != null) {
+      opts.outMimeType = URLConnection.guessContentTypeFromName("test." + image.mimeType);
```

Comment:
Can't this just be `opts.outMimeType = image.mimeType`? Aren't there just a couple of mime types possible? (e.g. image/png, image/jpeg, image/webp)?

Reply:
The `image.mimeType` uses the format likes `JPEG`, and it is not compatible with the current used mime type.